travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx

```diff
@@ -0,0 +1,288 @@
+'use client';
+
+import { RuleValidator } from '@/lib/validators/rule-validator/rule-validator';
+import { EmptyNullStringRule } from '@/lib/validators/rules/empty-null-string-rule';
+import { MatchingPasswordsRule } from '@/lib/validators/rules/matching-passwords-rule';
+import { PasswordComplexityRule } from '@/lib/validators/rules/password-complexity-rule';
+import { ValidEmailRule } from '@/lib/validators/rules/valid-email-rule';
+import { ValidNameRule } from '@/lib/validators/rules/valid-name-rule';
+
+import CloseIcon from '@mui/icons-material/Close';
+import GitHubIcon from '@mui/icons-material/GitHub';
+import GoogleIcon from '@mui/icons-material/Google';
+import {
+  Box,
+  Button,
+  Chip,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  IconButton,
+  Typography,
+  styled,
+  useTheme,
+} from '@mui/material';
+import { ChangeEventHandler, useState } from 'react';
+import { SignupFields } from './SignupFields/SignupFields';
+import { LoginFields } from './login-fields/LoginFields';
+
+/**
+ * Dialog used for signup and login
+ */
+
+interface AuthDialogProps {
+  open: boolean;
+  onDialogClose: () => void;
+  authDialogType: 'login' | 'signup';
+}
+
+export default function AuthDialog(props: AuthDialogProps) {
+  const [errors, setErrors] = useState<Record<string, string[]>>({});
```

Comment:
@LifeAsDev In the meeting you mentioned the case where the user tries to register with an e-mail that is already in use We should be able to add the back-end error response to the `errors` state object.

Reply:
yeah, what message I have to return from the backend?

Reply:
You could follow a similar structure and return an Record<string, string[]>. In the backend you have the message property return
{ email: ['Email already in use'] }
```